### ISSUE TITLE
discrepancy between pysam and samtools with secondary reads

### ISSUE THREAD

**Issue by Redmar-van-den-Berg** [pysam-developers/pysam#523]
I cannot get the `read.alignment.is_secondary` property to work as expected in pysam 0.11.2.2 (latest pip) and 0.10.0 (earlier pip).

When I analyse my reads with samtools (Version: 0.1.19-96b5f2294a), it reports 585 secondary alignments in the bamfile.
```
$ samtools view 21_untrimmed_100k.bam -f 256 | wc -l
585
```

However, when I write a simple pysam script to output the secondary alignments, I get no output.
Script:
```
#!/usr/bin/env python3
                                                                                                       
import sys                                                                                             
import pysam                                                                                           
                                                                                                       
print(pysam.__path__)                                                                                  
print(pysam.__version__)                                                                               
filename = sys.argv[1]                                                                                 
with pysam.AlignmentFile(filename, 'rb') as samfile:                                                   
    for pileupcolumn in samfile.pileup():                                                              
        for read in pileupcolumn.pileups:                                                              
            if read.alignment.is_secondary:                                                            
                print(read)
```
Using the latest version from pip
```
$ python3 secondary.py 21_untrimmed_100k.bam 
['/data/Experiment/redmar/Experiment_87/error_per_position/21/virt/lib/python3.5/site-packages/pysam']
0.11.2.2
```
Using an earlier version I had installed
```
$ python3 secondary.py 21_untrimmed_100k.bam 
['/home/redmar/.local/lib/python3.5/site-packages/pysam']
0.10.0
```

Am I doing something wrong here or is this an issue with pysam?

**Comment by AndreasHeger** [comment]
Thanks. The AlignmentFile.pileup() method applies several filters, can you try using the stepper="nofilter" option?

**Comment by Redmar-van-den-Berg** [comment]
Thanks for your response. If I add stepper="nofilter" as you suggested I find 27078 secondary reads. I guess the reason is that samtools also has some filters enabled by default. I was looking into this problem because I wanted to make sure that I did not count secondary reads twice, and I got worried when it seemed that pysam did not flag them.
Now that I know that they are filtered out by pileup() by default, I consider this issue close. Thanks for your help.
